cmake/common.cmake

```diff
@@ -15,6 +15,14 @@ function(llama_add_compile_flags)
 
             list(APPEND CXX_FLAGS -Wmissing-declarations -Wmissing-noreturn)
 
+            if (CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+                list(APPEND CXX_FLAGS -Wshadow)
+
+                if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+                    list(APPEND CXX_FLAGS -Wshadow-field-in-constructor)
+                endif()
```

Comment:
Decided to add the `-Wshadow-field-in-constructor` flag because there is no way to disable it with GCC, so it would be more consistent to have it in Clang as well. This requires to add underscore `_` suffix to constructor arguments that have the same name as the members in order to disambiguate the two.